python/foglamp/services/core/support.py

```diff
@@ -0,0 +1,253 @@
+
+__author__ = "Amarendra K Sinha"
+__copyright__ = "Copyright (c) 2017 OSIsoft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+
+_LOGGER = logger.setup(__name__)
+
+
+class SupportBuilder:
+    out_file_path = None
+    interim_file_path = None
+    base_url = None
+    headers = None
+
+    def __init__(self, support_dir="/tmp"):
+        try:
+            if not os.path.exists(support_dir):
+                os.makedirs(support_dir)
+            else:
+                self.check_and_delete_bundles(support_dir)
+
+            self.out_file_path = support_dir
+            self.interim_file_path = support_dir
+            self.base_url = 'http://localhost:8081/foglamp'
+            self.headers = {"Content-Type": 'application/json'}
+        except (OSError, Exception) as ex:
+            _LOGGER.error("Error in initializing SupportBuilder class: %s ", str(ex))
+            raise
+
+    async def build(self):
+        try:
+            loop = asyncio.get_event_loop()
+
+            today = datetime.datetime.now()
+            file_spec = today.strftime('%Y%m%d-%H-%M-%s')
+            tar_file_name = self.out_file_path+"/"+"support-{}.tar.gz".format(file_spec)
+            pyz = tarfile.open(tar_file_name, "w:gz")
+            try:
+                self.add_syslog_foglamp(pyz, file_spec)
+                self.add_syslog_storage(pyz, file_spec)
+                await self.add_table_configuration(pyz, file_spec, loop=loop)
+                await self.add_table_audit_log(pyz, file_spec, loop=loop)
+                await self.add_table_schedules(pyz, file_spec, loop=loop)
+                await self.add_table_scheduled_processes(pyz, file_spec, loop=loop)
+                await self.add_service_registry(pyz, file_spec, loop=loop)
+                self.add_machine_resources(pyz, file_spec)
+                self.add_psinfo(pyz, file_spec)
+            finally:
+                pyz.close()
+        except Exception as ex:
+            _LOGGER.error("Error in creating Support zip: %s ", str(ex))
+            raise
+        return tar_file_name
+
+    def check_and_delete_bundles(self, support_dir='/home/asinha/Development/FogLAMP/data/tmp/support'):
+        files = glob.glob(support_dir + "/" + "support*.tar.gz")
+        files.sort(key=os.path.getmtime)
+        if len(files) >= 3:
+            for f in files[:-2]:
+                if os.path.isfile(f):
+                    os.remove(os.path.join(support_dir, f))
+
+        # Delete all non *.tar.gz files
+        for f in os.listdir(support_dir):
+            if not fnmatch.fnmatch(f, 'support*.tar.gz'):
+                os.remove(os.path.join(support_dir, f))
+
+    def add_syslog_foglamp(self, pyz, file_spec):
+        # The foglamp entries from the syslog file
+
+        temp_file = self.interim_file_path + "/" + "syslog-{}".format(file_spec)
+        logs = logs_main()
+        data = {
+            "About": "Syslog for FogLAMP server",
+            "foglamp server logs": logs
```

Comment:
should not keys name be better? About -> about ? `foglamp server logs` > `foglampSyslog`